docs/current/guides/723462-use-secrets.md

```diff
@@ -298,6 +298,10 @@ secret env data: *** || secret file data:
 ***
 ```
 
+::danger
+Host environment variables containing secrets should always be declared using `WithSecretVariable()` and never using `WithEnvVariable()`. The former guarantees that the secret will not be visible in the container image history or layers.
```

Comment:
It's more than any secret that should be loaded as an env var by an app in a container should load it from a `withSecretVariable` env variable. If it is injected as a `withEnvVariable`, then the value of the env variable will be leaked in the build history of the container.